src/passes/CoalesceLocals.cpp

```diff
@@ -440,7 +440,7 @@ void CoalesceLocals::applyIndices(std::vector<Index>& indices, Expression* root)
   auto numParams = getFunction()->getNumParams();
   Index newNumLocals = 0;
   for (auto index : indices) {
-    newNumLocals = std::max(newNumLocals, index + 1);
+    newNumLocals = std::max((Index::value_type)newNumLocals, index + 1);
```

Comment:
is there no solution for this? doing `std:max` or `min` on `Index`es might not be that rare.